packages/core/src/internal-components/overlay/overlay.element.spec.ts

```diff
@@ -15,6 +15,14 @@ describe('Overlay helper functions: ', () => {
       expect(isNestedOverlay('ohai_2', 'ohai_', ['ohai_1', 'ohai_2', 'ohai_3'])).toBe(true, 'middle one checks');
     });
 
+    // this situation happens during an update loop when the focus trap list is updated just
+    // prior to the rest of the component and the component still needs to think of itself as
+    // "nested" or "layered". a good example is running an exit animation asynchronously
+    // outside of the update loop.
```

Comment:
This comment explains a need for a re-factoring of the focus traps. Animations revealed a bad state they could get into.